.github/workflows/nubots.yaml

```diff
@@ -0,0 +1,171 @@
+# Continuous Integration tests
+name: NUbots CI
+
+# Controls when the action will run.
+on:
```

Comment:
Could we break this up into [multiple files](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run) and use [path matching](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths)? So we don't have to build the docker image when we do a change to a cpp file, or build the codebase when we change a b tool. Probably for a future PR.

Reply:
We could even send a [slack message](https://github.com/archive/github-actions-slack) on push to master when the dockerfile is changed. :thinking:

Reply:
Great ideas. I'll make an issue for github actions next steps, like the documentation generation next steps one